### DISCUSSION TITLE
elastalert-server docker logs - ERROR elastalert: elastalert: Error finding recent pending alerts: NotFoundError 404

### DISCUSSION THREAD

**Discussion by pratikshatiwari** [jertel/elastalert2#1158] [Q&A] [answered]
Hi,

I have installed the elastalert-server2 and it is running too but in error log i can frequently see below message which is blocking to work elastalert-server & elastalert together.

Installed Python

python 3.9

https://github.com/jertel/elastalert2/
https://github.com/Karql/elastalert2-server

Error log as below:

17:16:42.012Z WARN elastalert: elasticsearch: POST http://localhost:9201/elastalert_status/_search?size=1000 [status:404 request:0.003s]
17:16:42.012Z ERROR elastalert: elasticsearch: POST http://localhost:9201/elastalert_status/_search?size=1000 [status:404 request:0.003s]
17:16:42.013Z ERROR elastalert: elastalert: Error finding recent pending alerts: NotFoundError(404, 'index_not_found_exception', 'no such index [elastalert_status]', elastalert_status, index_or_alias) {'query': {'bool': {'must': {'query_string': {'query': '!exists:aggregate_id AND alert_sent:false'}}, 'filter': {'range': {'alert_time': {'from': '2023-04-09T17:16:42.008720Z', 'to': '2023-04-11T17:16:42.008752Z'}}}}}, 'sort': {'alert_time': {'order': 'asc'}}} Traceback (most recent call last): File "/opt/elastalert/elastalert/elastalert.py", line 1476, in find_recent_pending_alerts res = self.writeback_es.search(index=self.writeback_index, body=query, size=1000) File "/usr/lib/python3.10/site-packages/elasticsearch-7.10.1-py3.10.egg/elasticsearch/client/utils.py", line 152, in _wrapped return func(*args, params=params, headers=headers, **kwargs) File "/usr/lib/python3.10/site-packages/elasticsearch-7.10.1-py3.10.egg/elasticsearch/client/init.py", line 1658, in search return self.transport.perform_request( File "/usr/lib/python3.10/site-packages/elasticsearch-7.10.1-py3.10.egg/elasticsearch/transport.py", line 392, in perform_request raise e File "/usr/lib/python3.10/site-packages/elasticsearch-7.10.1-py3.10.egg/elasticsearch/transport.py", line 358, in perform_request status, headers_response, data = connection.perform_request( File "/usr/lib/python3.10/site-packages/elasticsearch-7.10.1-py3.10.egg/elasticsearch/connection/http_requests.py", line 199, in perform_request self._raise_error(response.status_code, raw_data) File "/usr/lib/python3.10/site-packages/elasticsearch-7.10.1-py3.10.egg/elasticsearch/connection/base.py", line 315, in _raise_error raise HTTP_EXCEPTIONS.get(status_code, TransportError)( elasticsearch.exceptions.NotFoundError: NotFoundError(404, 'index_not_found_exception', 'no such index [elastalert_status]', elastalert_status, index_or_alias)

**Comment by jertel** [comment]
This project does not support elastalert2-server. As far as the 404 error it appears the indices did not get created. Did you follow the ElastAlert 2 [installation instructions](https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#setting-up-elasticsearch)?

**Comment by pratikshatiwari** [comment]
Hello @jertel 

Thank you for your responce, yes the indexes are present created by following elastalert-create-index as per instruction 
 https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#setting-up-elasticsearch (Python Package installation method)

Also wanted to user elastalert-kibana plugin to manage the rules from kibana hence i installed elastalert2-server, is there any way i can use elastalert-kibana-plugin by using elastalert2 https://github.com/jertel/elastalert2 


elastalert Index details as below, not sure why it is coming 404 error
**curl localhost:9201/_cat/indices/elastalert***
yellow open elastalert_status_status jWJdnw4sTMmgcFQDQKnyeQ 1 1 0 0 225b 225b
yellow open elastalert_status -L13Nm1VTi2iVgvpfjNSug 1 1 0 0 225b 225b
yellow open elastalert_status_past f2JNYQqpSmu6kKaxJFzoCg 1 1 0 0 225b 225b
yellow open elastalert_status_silence IGGSvzL3TqqQXK2PBjj6tw 1 1 0 0 225b 225b
yellow open elastalert_status_error WNCkLyLySsqyzLDNjcO3og 1 1 0 0 225b 225b

**Reply by jertel** [reply, marked as answer]
Try curling a POST manually to the URL in your original comment. If you get the 404 then you can move into the Elasticsearch logs to determine why the index appears to exist with one curl but not with the POST, and eliminate ElastAlert 2 as the source of the problem. Perhaps the fact that the index is in the yellow state has something to do with it.

I don't have any experience with the elastalert-kibana-plugin. There are others on this forum that might, so you'll have to wait to see if they have any insight.

**Reply by pratikshatiwari** [reply]
hi @jertel 

Thanks again, reason index are in yellow because of replica not available, when i use elastalaert alone with below method it works fine but i want to manage the rules through kibana do you suggest any way how i can manage elastalert rules through kibana like modifying/creating alert & alert monitoring

below working elastalert command without elastalert2-server

python -m elastalert.elastalert --verbose --rule examples/rules/example_frequency.yaml
